contracts/uniswap/KyberUniswapReserve.sol

```diff
@@ -97,8 +110,7 @@ contract KyberUniswapReserve is KyberReserveInterface, Withdrawable, Utils2 {
         // This makes the UNUSED warning go away.
         blockNumber;
 
-        require(isValidTokens(src, dest));
-
+        if (!isValidTokens(src, dest)) return 0;
         if (!tradeEnabled) return 0;
```

Comment:
add here. if srcQty == 0 return 0

srcQty 0 can happen with token to token,
if u don't fix. at least make sure it doesn't revert (like my code does... :()

Reply:
Added a check for it. Shouldn't this be handled by the KyberNetwork caller?
Also, is this required for trade() as well?

Reply:
for trade not required. for trade its "OK" to revert.
and if we return 0 trade shouldn't be called.

this is not handled when token to token and first toekn to eth returns 0 qty since can't do the trade.